Drop old capture loading.

florian-kuebler opened the pull request:
Old captures are not supported due to the changes on manual
instrumentation.

So let's drop all related code.

Bug: http://b/196374661
Test: Try to load an old and a new capture.

florian-kuebler commented:
> Can you clarify what "old" here means? What support are we breaking and what exactly does that mean for end-users? I thought we were supporting both the old and new manual instrumentation for a while.

The format that is based on the client protos in `capture_data.proto`. The new format is the one based on `CaptureFile` and contains the proto messages that come from the service (`capture.proto`).

florian-kuebler commented:
We dropped support for writing in this format a couple of releases ago, but kept support for reading, basically as it was still possible.

dimitry- commented on src/OrbitGl/App.cpp:
```diff
@@ -1189,9 +1188,8 @@ Future<ErrorMessageOr<CaptureListener::CaptureOutcome>> OrbitApp::LoadCaptureFro
-      load_result = orbit_client_model::capture_deserializer::Load(
-          file_path, this, module_manager_.get(), &capture_loading_cancellation_requested_);
+    } else {  // The old format is not supported anymore.
+      load_result = ErrorMessage("Old captures are not supported anymore.");
```
We should probably use capture_file_or_error.error() here because it provides more context.

florian-kuebler replied:
Makes sense. Done!

ronaldfw commented:
> We dropped support for writing in this format a couple of releases ago, but kept support for reading, basically as it was still possible.

Ok, so this is the very first, somewhat slow version of manual instrumentation. Then I don't have concerns with removing this.

florian-kuebler commented:
> > We dropped support for writing in this format a couple of releases ago, but kept support for reading, basically as it was still possible.
> 
> Ok, so this is the very first, somewhat slow version of manual instrumentation. Then I don't have concerns with removing this.

You mean, the very first version of serialization? The change itself has nothing to do with manual instrumentation. It was only triggered by the fact, that we now dropped support for reading the old manual instrumentation events.

dimitry- commented on src/OrbitGl/App.cpp:
```diff
@@ -1189,9 +1188,8 @@ Future<ErrorMessageOr<CaptureListener::CaptureOutcome>> OrbitApp::LoadCaptureFro
-    } else {  // Fall back to old capture format.
-      load_result = orbit_client_model::capture_deserializer::Load(
-          file_path, this, module_manager_.get(), &capture_loading_cancellation_requested_);
+    } else {  // The old format is not supported anymore.
```
nit: remove this comment? the error might be about not just the old format. could be an invalid capture file.

dimitry- reviewed (approved):
otherwise lgtm
